source/MaterialXRuntime/Private/PvtPrim.h

```diff
@@ -352,8 +355,15 @@ class PvtPrimSpec : public RtPrimSpec
         return _primAttr.find(name);
     }
 
+    const RtAttributeSpecVec& getAttributes() const override
+    {
+        return _primAttr._vec;
+    }
+
     const RtAttributeSpec* getPortAttribute(const RtPort& port, const RtToken& name) const override;
 
+    RtAttributeSpecVec getPortAttributes(const RtPort& port) const override;
```

Comment:
These methods return an std::vector now. I will change into an iterator instead when I have more time, to avoid the STL usage. In the port attribute case this is a bit complicated so don't have the bandwidth to solve that right now.

Reply:
But if we use the iterator pattern:
```
for (auto it : spec->getAttributes())
{
   ...
}
```
we don't need to update client code later when the API cahnges to an iterator.